packages/models/__tests__/store/store.ts

```diff
@@ -40,125 +60,304 @@ Reflect.defineMetadata(ProviderKey.Actor, { ref }, JSONStore)
 Reflect.defineMetadata(ProviderKey.Actor, { ref }, CustomStore)
 Reflect.defineMetadata(ProviderKey.Actor, { ref }, NoInstanceCustomStore)
 
+const createCell = ({
+  lock,
+  data,
+  type,
+}: {
+  type?: Script
+  lock?: Script
+  data?: HexString
+} = {}): Cell => {
+  return {
+    cellOutput: {
+      capacity: '0x16b969d00',
+      lock: lock || {
+        codeHash: '0x9bd7e06f3ecf4be0f2fcd2188b23f1b9fcc88e5d4b65a8637b17723bbda3cce8',
+        hashType: 'type',
+        args: '0x',
+      },
+      type,
+    },
+    data: data || '0x',
+    outPoint: {
+      txHash: `0x${'0'.repeat(64)}`,
+      index: '0x0',
+    },
+  }
+}
+
+const defaultOutpoint = `0x${'0'.repeat(64)}0`
+
 describe('test store', () => {
   describe('use json storage', () => {
+    describe('test init chain data', () => {
+      it('init with data', () => {
+        const store = new JSONStore<{ data: { a: BigNumber } }>({ data: true })
+        const chainData = store.initOnChain({ data: { a: BigNumber(20) } })
+        const expected = bytes.hexify(new JSONStorage().serialize({ a: BigNumber(20) }))
+        expect(chainData.data).toBe(expected)
+        expect(chainData.witness).toBeUndefined()
+        expect(chainData.lock).toBeUndefined()
+        expect(chainData.type).toBeUndefined()
+      })
+      it('init with witness', () => {
+        const store = new JSONStore<{ witness: { a: BigNumber } }>({ witness: true })
+        const chainData = store.initOnChain({ witness: { a: BigNumber(20) } })
+        const expected = bytes.hexify(new JSONStorage().serialize({ a: BigNumber(20) }))
+        expect(chainData.witness).toBe(expected)
+        expect(chainData.data).toBeUndefined()
+        expect(chainData.lock).toBeUndefined()
+        expect(chainData.type).toBeUndefined()
+      })
+      it('init with lock', () => {
+        const store = new JSONStore<{ lock: { args: BigNumber; codeHash: { schema: { a: string }; offset: 10 } } }>({
+          lock: { args: true, codeHash: { offset: 10 } },
+        })
```

Comment:
A question not related to function. is the schema defined reversed, code hash is expected to be a hash string while args could be deserialized to a structure, in actual use cases.

Reply:
Do you mean like this? https://github.com/yanguoyu/kuai/blob/feat-add-type/packages/models/__tests__/store/store.ts#L216-L257

Reply:
Yes, `args` consists of multiple fields in some cases, but `code hash` is definitely a hash string from `cell.typeScript` or `cell.data`.

Reply:
If `args` include multiple fields, developers can define multiple schema to create multiple `store`

Reply:
The point is that it's anti-intuitive to impose a scheme on code hash in CKB because code hash has a clear definition

> The hash of ELF formatted RISC-V binary that contains a CKB script. For space efficiency consideration, the actual script is attached to current transaction as a dep cell. Depending on the value of hash_type, the hash specified here should either match the hash of cell data part in the dep cell, or the hash of type script in the dep cell. The actual binary is loaded into an CKB-VM instance when they are specified upon the transaction verification.

It is a string from an unidirectional algorithm.

Ref: https://github.com/nervosnetwork/rfcs/blob/master/rfcs/0019-data-structures/0019-data-structures.md#description-1

Reply:
Got it, I will remove `code_hash` from `schema`.

Reply:
If so, the `lock` will only include `args`, I think I can use `lockArgs` for the schema defined.